offchain/danaswap-browser/flake-module.nix

```diff
@@ -0,0 +1,62 @@
+{ self, ... }:
+{
+  perSystem = { config, self', inputs', system, ... }:
+    let
+      pkgs = inputs'.nixpkgs.legacyPackages;
+      purs-nix = config.ps.purs-nix;
+      inherit (purs-nix) ps-pkgs;
+      inherit (config.ps) ctl-pkgs;
+      inherit (config) cat-lib offchain-lib;
+
+      danaswap-browser = {
+        ps =
+          purs-nix.purs
+            {
+              dependencies =
+                with ps-pkgs;
+                [
+                  halogen
+                  halogen-store
+                  ctl-pkgs.cardano-transaction-lib
+                  # self'.packages."offchain:danaswap-api"
+                ];
+              dir = ./.;
+            };
+        package =
+          let
+            nodeModules = pkgs.symlinkJoin {
+              name = "danaswap-browser-node-modules";
+              paths = [
+                config.ctl.nodeModules
+              ];
+            };
+          in
+          pkgs.runCommand "build-danaswap-browser" { }
+            # see buildPursProject: https://github.com/Plutonomicon/cardano-transaction-lib/blob/c906ead97563fef3b554320bb321afc31956a17e/nix/default.nix#L74
+            # see bundlePursProject: https://github.com/Plutonomicon/cardano-transaction-lib/blob/c906ead97563fef3b554320bb321afc31956a17e/nix/default.nix#L149
+            ''
+              export BROWSER_RUNTIME=1
+              cp -r ${danaswap-browser.ps.modules."DanaSwapBrowser.Home".output { }} homeOutput
+              cp -r ${danaswap-browser.ps.modules."DanaSwapBrowser.Pools".output { }} poolsOutput
+              cp ${./home.js} home.js
+              cp ${./pools.js} pools.js
+              cp -r ${nodeModules}/* .
+              export NODE_PATH="node_modules"
+              export PATH="bin:$PATH"
+              mkdir -p $out/dist
+              webpack --mode=production -c ${../webpack.config.js} -o $out/dist/home --entry ./home.js
+              webpack --mode=production -c ${../webpack.config.js} -o $out/dist/pools --entry ./pools.js
```

Comment:
Do you think it could make sense to make the `pool` and `home` separate package outputs?

Reply:
That way we could factor out one function which we can use to bundle both pool and home

Reply:
> Do you think it could make sense to make the `pool` and `home` separate package outputs?

yes, they will be imported by UI separately, hence separate package outputs

Reply:
> That way we could factor out one function which we can use to bundle both pool and home

@marijanp I did try to introduce a function `mkBundle`, 

```
      let
         mkBundle = module: outputDir: entryJs: distDir:
              let
                moduleFullPath = danaswap-browser.ps.modules."${module}".output { };
              in
              ''
                export BROWSER_RUNTIME=1
                cp -r ${moduleFullPath} ${outputDir}
                cp ${./${entryJs}} ${entryJs}
                cp -rp ${nodeModules}/* .
                export NODE_PATH="node_modules"
                export PATH="bin:$PATH"
                mkdir -p $out/dist
                webpack --mode=production -c ${../webpack.config.js} -o $out/dist/${distDir} --entry ./${entryJs}
              '';
            homeBundle = mkBundle "DanaSwap.Browser.Home" "homeOutput" "home.js" "home";
            poolsBundle = mkBundle "DanaSwap.Browser.Pools" "poolsOutput" "pools.js" "pools";
          in
          pkgs.runCommand "build-danaswap-browser" { } (lib.concatStrings [homeBundle poolsBundle]);
```

it will raise an error since copying `node_modules` to the same destination

```
error: builder for '/nix/store/50kg1pgfcfg4gcvhfbkz6f7iqc4ysld8-build-danaswap-browser.drv' failed with exit code 1;
       last 10 log lines:
       > cp: cannot remove './node_modules/assert/LICENSE': Permission denied
       > cp: cannot remove './node_modules/assert/build/assert.js': Permission denied
       > cp: cannot remove './node_modules/assert/build/internal/util/comparisons.js': Permission denied
       > cp: cannot remove './node_modules/assert/build/internal/errors.js': Permission denied
       > cp: cannot remove './node_modules/assert/build/internal/assert/assertion_error.js': Permission denied
       > cp: cannot remove './node_modules/parseurl/HISTORY.md': Permission denied
       > cp: cannot remove './node_modules/parseurl/package.json': Permission denied
       > cp: cannot remove './node_modules/parseurl/README.md': Permission denied
       > cp: cannot remove './node_modules/parseurl/LICENSE': Permission denied
       > cp: cannot remove './node_modules/parseurl/index.js': Permission denied
       For full logs, run 'nix log /nix/store/50kg1pgfcfg4gcvhfbkz6f7iqc4ysld8-build-danaswap-browser.drv'.
```

Reply:
yeah the problem is that you concat the string and this means that the script copies the node-modules twice which is in itself a little bit ugly. I'ld say lets leave it like that for now